pkg/mqtt-paho/Makefile.mqtt-paho

```diff
@@ -0,0 +1,10 @@
+MODULE = mqtt-paho
+
+RIOT_MQTT_IFACE_H = mqtt.h
+
+CFLAGS += -DMQTT_TASK=1
```

Comment:
The thing is that MQTTClient uses this flag to define MQTTTaskStart
```
#if defined(MQTT_TASK)
/** MQTT start background thread for a client.  After this, MQTTYield should not be called.
*  @param client - the client object to use
*  @return success code
*/
DLLExport int MQTTStartTask(MQTTClient* client);
#endif
```

Reply:
Just move this line to `pkg/mqtt--paho/Makefile.include` and it will be applied globally: no need to use it a second time in the application Makefile. Try it!